### PR TITLE
chore(deps): Move to go modules; Upgrade distribution to latest version.

### PR DESCRIPTION
**Pull request by jfuechsl**
This PR contains the following changes:
1. Move to go modules (from glide).
2. Update to the latest deis go base image.
3. Upgrade the distribution dependency to the github.com/teamhephy/distribution fork

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,7 +8,7 @@ else
 	GOOS=linux
 endif
 
-DEV_ENV_IMAGE := quay.io/deis/go-dev:0.11.1
+DEV_ENV_IMAGE := quay.io/deis/go-dev:v1.27.1
```
Comment (Cryptophobia) [inline Makefile]: Can you change to use our own image here: 

```
hephy/go-dev:v1.26.2
```

We are building it from here and merging upstream changes: https://github.com/teamhephy/docker-go-dev

**Review by Cryptophobia — CHANGES_REQUESTED** [review]
Looks real good and I would like to switch to go mods asap. Thanks @jfuechsl !

File: Makefile (hunk 1 above)
Comment (jfuechsl) [reply Makefile]: Pushed the change.

**Review by Cryptophobia — APPROVED** [review]
lgtm!

File: Makefile (hunk 1 above)
Comment (Cryptophobia) [reply Makefile]: Thank you for making this fix to use our own images @jfuechsl!

**Comment by jfuechsl** [comment]
Thanks!
I see that the cli binary is checked in directly in repos that use it (https://github.com/teamhephy/slugbuilder/blob/master/rootfs/Dockerfile#L33) vs. hosting it somewhere. Will this stay that way? I'm preparing the PRs for the builders and runners.

**Comment by Cryptophobia** [comment]
Good question @jfuechsl . That is done temporary because when we were making a new release the objstorage cli all of a sudden disappeared from the old deis' google storage link and I needed a quick fix.

This is where it will reside in the future when we make a new version: 
https://storage.googleapis.com/hephy-obj-storage-cli/bb8e054/objstorage

For example we can make one that is always `latest` tag for the cli: https://storage.googleapis.com/hephy-obj-storage-cli/latest/objstorage

**Pull request merged**